test/suites/examples/install-karpenter.yaml

```diff
@@ -0,0 +1,115 @@
+
+        ROLE="    - rolearn: arn:aws:iam::$(params.account-id):role/KarpenterNodeRole-$(params.cluster-name)\n      username: system:node:{{EC2PrivateDNSName}}\n      groups:\n      - system:nodes\n      - system:bootstrappers"
+        kubectl get -n kube-system configmap/aws-auth -o yaml | awk "/mapRoles: \|/{print;print \"$ROLE\";next}1" > /tmp/aws-auth-patch.yml
+        kubectl patch configmap/aws-auth -n kube-system --patch "$(cat /tmp/aws-auth-patch.yml)"
+
+  - name: irsa
+    image: docker.io/weaveworks/eksctl:0.86.0
+    script: |
+        eksctl create iamserviceaccount \
+          --cluster "$(params.cluster-name)" --name karpenter --namespace karpenter \
+          --role-name "$(params.cluster-name)-karpenter" \
+          --attach-policy-arn "arn:aws:iam::$(params.account-id):policy/KarpenterControllerPolicy-$(params.cluster-name)" \
+          --role-only \
+          --approve
+
+  - name: spot-slr
+    image: amazon/aws-cli
+    script: |
+        aws iam create-service-linked-role --aws-service-name spot.amazonaws.com || true
+
+  - name: helm-install-karpenter
+    image: alpine/helm:latest
+        export KARPENTER_IAM_ROLE_ARN="arn:aws:iam::$(params.account-id):role/$(params.cluster-name)-karpenter"
+        export CLUSTER_ENDPOINT="$(aws eks describe-cluster --name ${CLUSTER_NAME} --query "cluster.endpoint" --output text)"
+
+        helm repo add karpenter https://charts.karpenter.sh/
+        helm repo update
+
+        helm upgrade --install --namespace karpenter --create-namespace \
+          karpenter karpenter/karpenter \
+          --version $(params.karpenter-version) \
+          --set serviceAccount.annotations."eks\.amazonaws\.com/role-arn"=${KARPENTER_IAM_ROLE_ARN} \
+          --set clusterName=$(params.cluster-name) \
+          --set clusterEndpoint=${CLUSTER_ENDPOINT} \
+          --set aws.defaultInstanceProfile=KarpenterNodeInstanceProfile-$(params.cluster-name) \
+          --wait
+
+---
+apiVersion: tekton.dev/v1beta1
+kind: Pipeline
+metadata:
+  name: install-karpenter-pipeline
+  namespace: tekton
+spec:
+  tasks:
+  - name: install-karpenter
+    taskRef:
+      name: install-karpenter-task
+    workspaces:
+      - name: output
+        workspace: ws
+    params:
+    - name: cluster-name
+      value: "testing-ci-cluster"
+    - name: karpenter-version
+      value: "v0.11.0"
+    - name: account-id
+      value: "757543820853"
+    - name: pre-install-image
+      value: "docker.io/njtran/scratch:latest"
+    - name: aws-region
+      value: "us-west-2"
+  workspaces:
+  - name: ws
+    description: The folder where we write the message to
+---
+apiVersion: tekton.dev/v1beta1
+kind: PipelineRun
+metadata:
+  name: install-karpenter-pipelinerun
+  namespace: tekton
+spec:
+  pipelineRef:
+    name: install-karpenter-pipeline
+  podTemplate:
+    nodeSelector:
+      kubernetes.io/arch: amd64
+  workspaces:
+    - name: ws # this workspace name must be declared in the Pipeline
```

Comment:
Can you elaborate on this? Why does it have to be declared.

Reply:
This is a tekton specific detail. It's just to how Tekton works.